PennMobile/General/UserDefaults + Helpers.swift

```diff
@@ -398,6 +399,40 @@ extension UserDefaults {
     }
 }
 
+// MARK: - Penn Course Alert Settings
+extension UserDefaults {
+    
+    // MARK: Get and Save Preferences
+    // Set values for each PCA option
+    func set(_ PCAOption: PCAOption, to newValue: Bool) {
+        var prefs = getAllPCAPreferences()
+        prefs[PCAOption.rawValue] = newValue
+        saveAll(PCAPreferences: prefs)
+    }
+    
+    // Get values for each PCA option (default to false if no preference exists)
+    func getPreference(for option: PCAOption) -> Bool {
+        let prefs = getAllPCAPreferences()
+        return prefs[option.rawValue] ?? false
+    }
+    
+    // Fetch preferences from disk
+    func getAllPCAPreferences() -> PCAPreferences {
+        let decoder = JSONDecoder()
+        if let decodedData = UserDefaults.standard.data(forKey: UserDefaultsKeys.PCAPreferences.rawValue) {
+            return (try? decoder.decode(PCAPreferences.self, from: decodedData)) ?? .init()
+        }
+        return .init()
```

Comment:
I don't think that should ever happen, but I have tested changing it on the backend while having the page open. Nothing happens until you refresh or leave, in which case it will update with the new settings.